pulsar-rpc-contrib/src/main/java/org/apache/pulsar/rpc/contrib/client/PulsarRpcClient.java

```diff
@@ -0,0 +1,180 @@
+        ConcurrentHashMap<String, CompletableFuture<V>> pendingRequestsMap = new ConcurrentHashMap<>();
+        MessageDispatcherFactory<T, V> dispatcherFactory = new MessageDispatcherFactory<>(
+                client,
+                builder.getRequestSchema(),
+                builder.getReplySchema(),
+                builder.getReplySubscription());
+
+        Producer<T> producer = dispatcherFactory.requestProducer(builder.getRequestProducer());
+        RequestSender<T> sender = new RequestSender<>(null == builder.getReplyTopic()
+                ? builder.getReplyTopicsPattern().pattern() : builder.getReplyTopic());
+
+        RequestCallBack<V> callBack = builder.getCallBack() == null ? new DefaultRequestCallBack<>()
+                : builder.getCallBack();
+
+        ReplyListener<V> replyListener = new ReplyListener<>(pendingRequestsMap, callBack);
+        Consumer<V> consumer = dispatcherFactory.replyConsumer(
+                builder.getReplyTopic(),
+                replyListener,
+                builder.getReplyTopicsPattern(),
+                builder.getPatternAutoDiscoveryInterval());
+
+        return new PulsarRpcClient<>(
+                pendingRequestsMap,
+                builder.getReplyTimeout(),
+                sender,
+                producer,
+                consumer,
+                callBack);
+    }
+
+    /**
+     * Creates a builder for configuring a new {@link PulsarRpcClient}.
+     *
+     * @param requestSchema The schema for request messages.
+     * @param replySchema The schema for reply messages.
+     * @return A new instance of {@link PulsarRpcClientBuilder}.
+     */
+    public static <T, V> PulsarRpcClientBuilder<T, V> builder(
+            @NonNull Schema<T> requestSchema, @NonNull Schema<V> replySchema) {
+        return new PulsarRpcClientBuilder<>(requestSchema, replySchema);
+    }
+
+    /**
+     * Closes this client and releases any resources associated with it. This includes closing any active
+     * producers and consumers and clearing pending requests.
+     *
+     * @throws PulsarClientException if there is an error during the closing process.
+     */
+    @Override
+    public void close() throws PulsarClientException {
+        try (requestProducer; replyConsumer) {
+            pendingRequestsMap.forEach((correlationId, future) -> {
+                future.cancel(false);
+            });
+            pendingRequestsMap.clear();
+        }
+    }
+
+    /**
+     * Synchronously sends a request and waits for the response.
+     *
+     * @param correlationId A unique identifier for the request.
+     * @param message The message builder used to build and send the request.
+     * @return The reply value.
+     * @throws Exception if an error occurs during the request or while waiting for the reply.
+     */
+    public V request(String correlationId, TypedMessageBuilder<T> message) throws Exception {
```

Comment:
We'd better not expose the MQ implementation. Only key and value are passed in here.
This will make it easier for users to use.

Reply:
> We'd better not expose the MQ implementation. Only key and value are passed in here. This will make it easier for users to use.

It is also considered here that some custom configurations may be made for the `property` or `eventTime` of the message. It is a bit too inflexible to pass only key and value. It may be better for users to construct a message passed in by themselves.

Reply:
It allows users to pass in config. As I said before, you can load more configuration files for Producer and Consumer through `.loadConf()`. You can also pass in message configuration through `message.loadConf()`.

Reply:
Each message and the request producer has been modified using the loadConf().